iceoryx-sys/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+#![recursion_limit = "256"]
+
+#[macro_use]
+extern crate cpp;
+
+pub mod introspection;
+
+mod publisher;
+pub use publisher::Publisher;
+
+mod publisher_options;
+pub use publisher_options::PublisherOptions;
+
+mod queue_policy;
+pub use queue_policy::ConsumerTooSlowPolicy;
+pub use queue_policy::QueueFullPolicy;
+
+mod runtime;
+pub use runtime::Runtime;
+
+mod subscriber;
+pub use subscriber::ConditionVariable;
+pub use subscriber::SubscribeState;
+pub use subscriber::Subscriber;
+pub use subscriber::SubscriberArc;
+pub use subscriber::SubscriberRc;
+pub use subscriber::SubscriberStrongRef;
+pub use subscriber::SubscriberWeakRef;
```

Comment:
Why do we have a strong and a weak ref subscriber?

Reply:
It's quite long since I wrote this code and only God and I understood what it was for. Now only God knows.
Seriously, the `Sample` holds an [`Rc`](https://doc.rust-lang.org/std/rc/struct.Rc.html) or [`Arc`](https://doc.rust-lang.org/std/sync/struct.Arc.html) owning reference to the subscriber. The idea was to downgrade this reference to a non-owning [`Weak for Rc`](https://doc.rust-lang.org/std/rc/struct.Weak.html) or [`Weak for Arc`](https://doc.rust-lang.org/std/sync/struct.Weak.html) reference. But I'm not sure anymore why.